### ISSUE TITLE
Windows support

### ISSUE THREAD

**Issue by kakashi-hadakei** [gphoto/gphoto2#85]
Is there a windows build of gphoto2?

**Comment by kakashi-hadakei** [comment]
I need a version higher than 2.4

**Comment by msmeissn** [comment]
perhaps see https://github.com/gphoto/libgphoto2/issues/33

**Comment by peterbud** [comment]
Yes, there is a native Windows port, you can use them with MSYS2/MINGW
[libgphoto2](https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-libgphoto2)
[gphoto2](https://github.com/Alexpux/MINGW-packages/tree/master/mingw-w64-gphoto2)

**Comment by vividos** [comment]
I tried this a while back, using Visual Studio and a self written compat library:
https://github.com/vividos/libgphoto2/releases
I can't say if I done everything right, as I never used the API very much. Also only PTP2 and Canon camlibs were ported.

I recently tried another approach, which you can find here:
https://github.com/vividos/RemotePhotoTool/tree/master/build
You just need the .cmd file and the msvc folder. The .cmd file downloads the latest release, copies over the msvc folder and starts a Visual Studio build. Again, only contains the base libraries and the PTP2 and Canon camlibs, but porting other camlibs should be easy.

**Comment by define-private-public** [comment]
I was able to build libgphoto2 from source (using the MSYS2 environment & GCC), but I've been having segfaults when trying to call `gp_port_info_list_new()`, which is the first libgphoto2 function in my program.  I'm using the testing code form over here: https://github.com/gphoto/libgphoto2/issues/208#issuecomment-392191848

Should I be using a different compiler or toolchain on Windows?

**Comment by cgilles** [comment]
Hi all,

I take a look how gphoto2 can be used under Windows natively. The cross-compilation or MSYS2/MINGW tool-chain is not the right way to go, as it requires to replace the Windows USB layer by libUSB which is not acceptable at all.

A native way as RemotePhotoTool project do will be better so far, but of course with all current camera supported by Gphoto2 under Linux, not only the Canon devices. See also the Microsoft/VCPKG comment here : https://github.com/microsoft/vcpkg/issues/5075#issuecomment-633519224

So my Q is : there is plan to do it ?

Another Q is about to port current Gphoto2 build scripts to CMake, which will be more portable, modern, and easy to maintain.

Best regards

Gilles Caulier
